service/history/nDCHistoryReplicator.go

```diff
@@ -954,9 +956,7 @@ BackfillLoop:
 	return lastEventTime, nil
 }
 
-func copyAndSortAncestors(input []*persistencespb.HistoryBranchRange) []*persistencespb.HistoryBranchRange {
-	ans := make([]*persistencespb.HistoryBranchRange, len(input))
-	copy(ans, input)
+func sortAncestors(ans []*persistencespb.HistoryBranchRange) {
 	if len(ans) > 0 {
```

Comment:
Do you really need this check? I think `sort.Slice` should work fine with both nil and empty sliced.